.github/workflows/test-integration-rosa.yaml

```diff
@@ -0,0 +1,359 @@
+          echo "kubeconfig_raw=${encrypted_kubeconfig_base64}" >> "$GITHUB_OUTPUT"
+
+      ## Write for matrix outputs workaround
+      - uses: cloudposse/github-action-matrix-outputs-write@ed06cf3a6bf23b8dce36d1cf0d63123885bb8375 # v1
+        id: out
+        with:
+          matrix-step-name: ${{ github.job }}
+          matrix-key: ${{ matrix.distro.name }}
+          outputs: |-
+            kubeconfig_raw: ${{ steps.export_kube_config.outputs.kubeconfig_raw }}
+
+  read_kube_configs:
+    name: "Read kube configs from matrix"
+    runs-on: ubuntu-latest
+    needs: [ prepare_clusters ]
+    steps:
+      - uses: cloudposse/github-action-matrix-outputs-read@33cac12fa9282a7230a418d859b93fdbc4f27b5a # v1
+        id: read
+        with:
+          matrix-step-name: prepare_clusters
+    outputs:
+      kubeconfig: "${{ steps.read.outputs.result }}"
+
+  launch_tests:
+    name: "Launch tests"
+    runs-on: ubuntu-latest
+    needs:
+      - define_matrix
+      - read_cluster_vars
+      - read_kube_configs
+    strategy:
+      fail-fast: false
+      matrix:
+        distro: ${{ fromJson(needs.define_matrix.outputs.platform_matrix) }}
+        flow: ${{ fromJson(needs.define_matrix.outputs.flow_matrix) }}
+    steps:
+      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4
+        with:
+          fetch-depth: 1
+          ref: ${{ inputs.camunda-helm-git-ref }}
+
+      - name: Set up Kubeconfig
+        run: |
+          mkdir -p $HOME/.kube
+          echo "${{ fromJson(needs.read_kube_configs.outputs.kubeconfig).kubeconfig_raw[matrix.distro.name] }}" | base64 --decode > encrypted_kubeconfig.enc
+          openssl enc -aes-256-cbc -d -in encrypted_kubeconfig.enc -out "$HOME/.kube/config" -pass pass:"${GITHUB_TOKEN}" -pbkdf2
+          rm encrypted_kubeconfig.enc
+          chmod 600 $HOME/.kube/config
+
+      - name: Ensure identifier is set and append cluster name
+        run: |
+          INPUTS_IDENTIFIER="${{ env.INPUTS_IDENTIFIER }}"
+          if [ -z "$INPUTS_IDENTIFIER" ]; then
+            INPUTS_IDENTIFIER="$(git rev-parse --short ${{ github.sha }})"
+          fi
+
+          echo "INPUTS_IDENTIFIER=$INPUTS_IDENTIFIER-${{ fromJson(needs.read_cluster_vars.outputs.result).cluster_name[matrix.distro.name] }}" | tee -a "$GITHUB_ENV"
+
+      - name: Launch tests
+        uses: ./.github/actions/chart-test-recipes
+        with:
+          platform: "${{ matrix.distro.platform }}"
+          type: "${{ matrix.distro.type }}"
+
+          identifier: "${{ env.INPUTS_IDENTIFIER }}"
+          persistent: "${{ env.TEST_PERSISTENT }}"
+          flow: "${{ matrix.flow }}"
+          test-enabled: "${{ env.INPUTS_TEST_ENABLED }}"
+
+          caller-git-ref: ${{ env.INPUTS_CALLER_GIT_REF }}
+          extra-values: ${{ env.INPUTS_EXTRA_VALUES }}
+          ingress-hostname-base: ${{ env.TEST_HOSTNAME_BASE }}
+
+          gh_app_id: ${{ secrets.GH_APP_ID_DISTRO_CI_MANAGE_GH_ENVS }}
```

Comment:
Let's use the token directly instead of the app (any token with the correct permissions should work).

Reply:
> extracted from the composite action [...] and re-integrated in the workflow [.github/workflows/test-integration-rosa.yaml](https://github.com/camunda/camunda-platform-helm/pull/1816/files/7cb049313baa19d14ac07c7884eeb5b4e6e0369d#diff-87e1b9424eb85d9dd7f6cbd995d57a09d1b8eff940c534f3406e0b4f5b51200d)

yes exactly, so we just pass the token (which will be generated using the `tibdex/github-app-token` outside the composition).